### PR TITLE
feat: compare List.ofFn and Array.ofFn

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Batteries/Data/Array/OfFn.lean [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Array
+
+/-! ### ofFn -/
+
+@[simp]
+theorem data_ofFn (f : Fin n → α) : (ofFn f).data = List.ofFn f := by
+  ext1
+  rw [List.getElem?_eq]
+  simp [ofFnNthVal]
```
Comment (mattrobball) [inline Batteries/Data/Array/OfFn.lean:18]:
```suggestion
  simp only [getElem?_eq, data_length, size_ofFn, length_ofFn, getElem_ofFn]
```

File: Batteries/Data/List/OfFn.lean [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+@[simp]
+theorem length_ofFn (f : Fin n → α) : (ofFn f).length = n := by
+  simp only [ofFn]
+  induction n with
+  | zero => simp
+  | succ n ih => simp [Fin.foldr_succ, ih]
+
+@[simp]
+theorem getElem_ofFn (f : Fin n → α) (i : Nat) (h : i < (ofFn f).length) :
```
Comment (mattrobball) [inline Batteries/Data/List/OfFn.lean:23]: It doesn't look like this is a concern currently in the code base but should this and `getElem?_ofFn` be `protected`?

**Review by mattrobball — APPROVED** [review]

File: Batteries/Data/List/OfFn.lean (hunk 2 above)
Comment (kim-em) [reply Batteries/Data/List/OfFn.lean:23]: Sure. Better done early.

**Review by mattrobball — APPROVED** [review]
Very nice!

**Pull request merged**